langchain4j-azure-open-ai/src/test/java/dev/langchain4j/model/azure/AzureOpenAiChatModelIT.java

```diff
@@ -130,7 +129,6 @@ void should_call_function_with_argument() {
         List<ChatMessage> chatMessages = new ArrayList<>();
         chatMessages.add(systemMessage);
         chatMessages.add(userMessage);
-        chatMessages.add(aiMessage);
```

Comment:
I have previously added `aiMessage` here on purpose. `toolExecutionResultMessage` is the response to `aiMessage`. In the real-world scenario `aiMessage` should be present, otherwise the model can get confused. Moreover, in the newer OpenAI "tools" api (which deprecates "functions" api, search for "tool" and "function" [here](https://platform.openai.com/docs/api-reference/chat/create)) each `toolExecutionResultMessage` should correspond to a tool call present in `aiMessage`. If you don't supply `aiMessage`, you will get an error like this:
```
{
    "error": {
        "message": "Invalid parameter: messages with role 'tool' must be a response to a preceeding message with 'tool_calls'.",
        "type": "invalid_request_error",
        "param": "messages.[0].role",
        "code": null
    }
}
```

Reply:
Thank you, I really messed up on this one! Let me check and try to understand the issue.

Reply:
So, the issue is that AiMessage can't have a text that is "null" now (the integration test will fail because the field is empty), so I'm forcing this to "" in https://github.com/jdubois/langchain4j/commit/1b97059c359dddb932f15ced0a76e9384b899ea0 - @langchain4j I believe this is correct, WDYT?

Reply:
@jdubois sorry, I am not sure I understand the problem. `AiMessage.text == null` is ok for the case where `AiMessage.toolExecutionRequests != null`. AiMessage can contain either `text` or `toolExecutionRequests`, but not both. There should be no empty `AiMessage.text`. I just did a [change](https://github.com/langchain4j/langchain4j/pull/370/files) to reflect these constraints, please check.

Reply:
My bad, the constructors in `AiMessage` were really confusing, hopefully now it is better. I was not paying much attention to ctors as I usually use static factory methods, there it was quite clear which inputs are required to create `AiMessage`

Reply:
@langchain4j so I migrated to it in https://github.com/langchain4j/langchain4j/pull/359/commits/39207e89d0768bf59b82d8b064be600fe7a5b91f but and that compiles, but the integration tests won't work.

I'm getting:
```json
com.azure.core.exception.HttpResponseException: Status code 400, "{
  "error": {
    "message": "'content' is a required property - 'messages.2'",
    "type": "invalid_request_error",
    "param": null,
    "code": null
  }
}
```

This is because the text is `null` and not empty. I agree this looks like an Azure OpenAI SDK issue, then the only way to solve this currently is to pass an empty String.

Reply:
Ah I see... Can we adapt mapping logic from langchain4j to azure messages and insert empty string if aiMessage.text is null?

Reply:
So I created a specific AiMessage implementation for Azure OpenAI in https://github.com/langchain4j/langchain4j/pull/359/commits/5ffc013d0e1b707097b2f8401a35bc1c2cff4ce5 - I'm not a huge fan of this code, but it does the job and doesn't pollute the core LangChain4J implementation, so I guess it's the best option.

Reply:
Yeah me too... this can be simplified to this change in `InternalAzureOpenAiHelper.toOpenAiMessage()`:
`new ChatRequestAssistantMessage(getOrDefault(message.text(), ""));` and you don't need a whole new class for this

Reply:
Thanks, done!